inc/zoo/swar/SWAR.h

```diff
@@ -74,57 +78,152 @@ template<int NBits, typename T = uint64_t> struct SWAR {
+    }
+
+    constexpr auto most() const noexcept {
+        return this->m_v & MostMask;
+    }
+
+    // Returns only the most significant bits at specified position.
+    constexpr auto most(u32 pos) const noexcept {
+        constexpr auto filter = (T(1) << LaneBits) - 1;
+        const auto keep = (filter << (LaneBits * pos)) & MostMask;
+        return this->m_v & keep;
+    }
+
+    // Returns only the most significant bits at specified position, 'decoded' to their integer value.
+    constexpr auto mostFlat(u32 pos) const noexcept {
+        return most(pos) >> (LaneBits*pos)>> NBitsLeast;
+    }
+
+    // Sets the lsb sublane at |pos| with least significant NBitsLeast of |in|
+    constexpr auto least(T in, int pos) const noexcept {
+        constexpr auto filter = (T(1) << LaneBits) - 1;
+        const auto keep = ~(filter << (LaneBits * pos)) | MostMask;
+        const auto rdyToInsert = this->m_v & keep;
+        const auto rval = rdyToInsert | ((in & LeastMask) << (LaneBits * pos));
+        return SL(rval);
+    }
+
+    // Sets the msb sublane at |pos| with least significant NBitsMost of |in|
+    constexpr auto most(T in, int pos) const noexcept {
+        constexpr auto filter = (T(1) << LaneBits) - 1;
+        const auto keep = ~(filter << (LaneBits * pos)) | LeastMask;
+        const auto rdyToInsert = this->m_v & keep;
+        const auto insVal = (((in<<NBitsLeast) & MostMask) << (LaneBits * pos));
+        const auto rval = rdyToInsert | insVal;
+        return SL(rval);
+    }
+};
+
+
+
 /// Defining operator== on base SWAR types is entirely too error prone. Force a verbose invocation.
 template<int NBits, typename T = uint64_t> constexpr auto horizontalEquality(SWAR<NBits, T> left, SWAR<NBits, T> right) {
-  return left.m_v == right.m_v;
+    return left.m_v == right.m_v;
 }
 
 /// Isolating >= NBits in underlying integer type currently results in disaster.
 // TODO(scottbruceheart) Attempting to use binary not (~) results in negative shift warnings.
 template<int NBits, typename T = uint64_t>
 constexpr auto isolate(T pattern) {
-  return pattern & ((T(1)<<NBits)-1);
+    return pattern & ((T(1)<<NBits)-1);
 }
 
-/// Clears the lowest bit set in type T
+/// Clears the least bit set in type T
 template<typename T = uint64_t>
 constexpr auto clearLSB(T v) {
-  return v & (v - 1);
+    return v & (v - 1);
 }
 
-/// Leaves on the lowest bit set, or all 1s for a 0 input.
+/// Leaves on the least bit set, or all 1s for a 0 input.
 template<typename T = uint64_t>
 constexpr auto isolateLSB(T v) {
-  return v & ~clearLSB(v);
+    return v & ~clearLSB(v);
 }
 
 template<int NBits, typename T = uint64_t>
-constexpr auto lowestNBitsMask() {
-  return (T(1)<<NBits)-1;
+constexpr auto leastNBitsMask() {
+    return (T(1)<<NBits)-1;
+  //return ~((~T(0))<<NBits);
+}
+
+template<int NBits, uint64_t T>
+constexpr auto leastNBitsMask() {
+  // return (u64(1)<<NBits)-1;
+    return ~((0ull)<<NBits);
```

Comment:
I prefer these to be constants instead of functions

Reply:
How do we make them constants for arbitrary types?